addons/mass_mailing_sms/models/mailing_mailing.py

```diff
@@ -51,14 +54,22 @@ def _compute_body_plaintext(self):
                 mailing.body_plaintext = mailing.sms_template_id.body
 
     @api.depends('mailing_trace_ids.failure_type')
-    def _compute_sms_has_insufficient_credit(self):
-        mailing_ids = self.env['mailing.trace'].sudo().search([
-            ('mass_mailing_id', 'in', self.ids),
-            ('trace_type', '=', 'sms'),
-            ('failure_type', '=', 'sms_credit')
-        ]).mapped('mass_mailing_id')
-        for mailing in self:
-            mailing.sms_has_insufficient_credit = mailing in mailing_ids
+    def _compute_sms_has_iap_failure(self):
+        failures = ['sms_acc', 'sms_credit'] 
+        traces = self.env['mailing.trace'].sudo().read_group([
+                    ('mass_mailing_id', 'in', self.ids),
+                    ('trace_type', '=', 'sms'),
+                    ('failure_type', 'in', failures)
+        ], ['mass_mailing_id', 'failure_type'], ['mass_mailing_id', 'failure_type'], lazy=False)
+
+        trace_dict = dict.fromkeys(self.ids, {key: False for key in failures})
+        for t in traces:
+            trace_dict[t['mass_mailing_id'][0]][t['failure_type']] =  t['__count'] and True or False
+
+        for mail in self:
+            mail.sms_has_insufficient_credit = trace_dict[mail.id]['sms_credit']
```

Comment:
Mmmh we have funny errors with that kind of use of ids / id in compute methods when creating a record. Don't you have issue like "trying to reach newID-12" ?

Anway you can do trace_dict.get(mail.id, {}).get('sms_credit', False) so that new IDs don't mess with your code.

Reply:
I didn't stumble upon that while testing here but I might have jumped a stepwhile testing...